src/mesh/data/gridfromoptions.cxx

```diff
@@ -11,50 +11,54 @@
 
 #include <unused.hxx>
 
-bool GridFromOptions::hasVar(const std::string &name) {
-  return options->isSet(name);
-}
+bool GridFromOptions::hasVar(const std::string& name) { return options->isSet(name); }
 
-bool GridFromOptions::get(Mesh *UNUSED(m), std::string &sval, const std::string &name) {
-  if(!hasVar(name)) {
-    sval = "";
+bool GridFromOptions::get(Mesh* UNUSED(m), std::string& sval, const std::string& name) {
+  if (!hasVar(name)) {
+    const std::string def{};
+    output_warn.write("Variable '%s' not in mesh options. Setting to \"%s\"\n",
+                      name.c_str(), def.c_str());
+    sval = def;
     return false;
   }
-  
+
   options->get(name, sval, "");
   return true;
 }
 
-bool GridFromOptions::get(Mesh *UNUSED(m), int &ival, const std::string &name) {
-  if(!hasVar(name)) {
-    ival = 0;
+bool GridFromOptions::get(Mesh* UNUSED(m), int& ival, const std::string& name) {
+  if (!hasVar(name)) {
+    constexpr int def{0};
+    output_warn.write("Variable '%s' not in mesh options. Setting to %d\n", name.c_str(),
+                      def);
+    ival = def;
     return false;
   }
-  
+
   options->get(name, ival, 0);
   return true;
 }
 
-bool GridFromOptions::get(Mesh *UNUSED(m), BoutReal &rval, const std::string &name) {
-  if(!hasVar(name)) {
-    rval = 0.0;
+bool GridFromOptions::get(Mesh* UNUSED(m), BoutReal& rval, const std::string& name) {
+  if (!hasVar(name)) {
+    constexpr BoutReal def{0.0};
+    output_warn.write("Variable '%s' not in mesh options. Setting to %e\n", name.c_str(),
+                      def);
+    rval = def;
     return false;
   }
-  
-  // Fetch expression as a string 
-  std::string expr;
-  options->get(name, expr, "0");
 
-  // Parse, and evaluate with x,y,z,t = 0
-  std::shared_ptr<FieldGenerator> gen = FieldFactory::get()->parse(expr, options);
+  auto expr = (*options)[name].withDefault(std::string{"0"});
```

Comment:
I think this can now be replaced with `rval = (*options)[name].withDefault(0.0);`
since `Options` now parses strings and generates values with `x=y=z=t=0`.

Reply:
Ah thanks, I thought there might be a nicer way! I'll change it shortly